CAIPs/caip-282.md

```diff
@@ -0,0 +1,385 @@
+          accounts: [
+            "eip155:1:0x43e3ca49c7be4f429abce408da6b738f879d02a0",
+            "eip155:10:0x43e3ca49c7be4f429abce408da6b738f879d02a0"
+          ]
+        },
+      },
+      sessionProperties: {
+        expiry: "2024-06-06T13:10:48.155Z",
+      }
+    }
+  },
+});
+
+// 4. Once the connection is established then the Wallet provider can receive
+// incoming CAIP-27 requests which will be prompted to the user to sign and
+// once signed the response is sent back to the dapp with the expected result
+const request = {}
+window.addEventListener("caip282:request:350670db-19fa-4704-a166-e52e178b59d2", (event) => {
+  request = event.data
+});
+// wallet provider publishes for reponses
+window.postMessage("message", {
+  event: "caip282:respond:350670db-19fa-4704-a166-e52e178b59d2",
+  data: {
+    id: request.id, // 2
+    jsonrpc: "2.0",
+    result: "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
+  },
+});
+```
+
+## Security Considerations
+
+The advantage of using window.postMessage over existing standards that leverage window.dispatchEvent is the prevention of prototype pollution but that still does not mean that there aren't existing attacks that must be considered:
+
+### Wallet Imitation and Manipulation
+
+Application developers are expected to actively detect for misbehavior of properties or functions being modified in order to tamper with or modify other wallets. One way this can be easily achieved is to look for when the uuid property within two Wallet Data objects match. Applications and Libraries are expected to consider other potential methods that the Wallet Data objects are being tampered with and consider additional mitigation techniques to prevent this as well in order to protect the user.
+
+### Prevent SVG Javascript Execution
+
+The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. This Javascript executes within the context of the page and can therefore modify the page or the contents of the page. So when considering the experience of rendering the icons, DApps need to take into consideration how they’ll approach handling these concerns in order to prevent an image being used as an obfuscation technique to hide malicious modifications to the page or to other wallets.
+
+## Privacy Considerations
+
+Any form of wallet discoverability must alwasys take in consideration wallet fingerprinting that can happen by malicious webpages or extensions that attempt to capture user information. Thus wallet providers can abstain from publishing announceWallet messages on every page load and wait for incoming promptWallet messages. Yet this open the possiblity for race conditions where wallet providers could be initialized after the promptWallet message was published therefore be undiscoverable. It is recommended that if wallet providers do offer this more "private connect" feature that is only enabled optionally by users rather than set by default.
```

Comment:
```suggestion
Any form of wallet discoverability must alwasys take in consideration wallet fingerprinting that can happen by malicious webpages or extensions that attempt to capture user information. Wallet providers can abstain from publishing `announceWallet` messages on every page load and wait for incoming `promptWallet` messages. Yet this opens the possibility for race conditions where wallet providers could be initialized after the `promptWallet` message was published and therefore be undiscoverable. It is recommended that wallet providers offer this more "private connect" feature that users only enable optionally, rather than set by default.
```